### PR TITLE
Adding Regression Evaluation Metrics

### LINKED ISSUE #1127: Evaluation metrics - Regression / Generative models
**Is your feature request related to a problem? Please describe.**
A list of the metrics to be implemented to be applied in the case of a regression/generative task with possible links for implementation / reference

- Mean absolute error [https://pytorch.org/docs/master/generated/torch.nn.L1Loss.html#torch.nn.L1Loss](url)
- Mean squared error [https://pytorch.org/docs/master/generated/torch.nn.MSELoss.html#torch.nn.MSELoss](url)
- Smooth L1 Loss [https://pytorch.org/docs/master/generated/torch.nn.SmoothL1Loss.html#torch.nn.SmoothL1Loss](url)
- Root MSE [](url)
- Peak Signal to Noise Ratio [https://github.com/tensorflow/tensorflow/blob/v2.2.0/tensorflow/python/ops/image_ops_impl.py#L3358](url)
- Structural Similarity Index Measure [https://github.com/tensorflow/tensorflow/blob/v2.2.0/tensorflow/python/ops/image_ops_impl.py#L3569](url)
- Multi-scale SSIM [https://github.com/tensorflow/tensorflow/blob/v2.2.0/tensorflow/python/ops/image_ops_impl.py#L3648](url
- Negative Guassian Log Likelihood [https://pytorch.org/docs/stable/nn.html#torch.nn.NLLLoss](url)
- Maximum Mean Discrepancy [https://github.com/ZongxianLee/MMD_Loss.Pytorch/blob/master/mmd_loss.py#L5](url)
- Inception Score [https://github.com/sbarratt/inception-score-pytorch](url)
- Frechet Inception Distance [https://github.com/xuqiantong/GAN-Metrics](url)

**Additional context**
This comes from the work on metrics by task type done by the Metrics Task Force

### PR DESCRIPTION
**Pull request by masadcv**
Signed-off-by: masadcv <muhammad.asad@kcl.ac.uk>

Partially Fixes #1127  . 

### Description
Proposal implementation for Regression Evaluation metrics. 

This pull request includes the following evaluation metrics:
- MSE: https://en.wikipedia.org/wiki/Mean_squared_error
- MAE: https://en.wikipedia.org/wiki/Mean_absolute_error
- RMSE: https://en.wikipedia.org/wiki/Root-mean-square_deviation
- PSNR: https://en.wikipedia.org/wiki/Peak_signal-to-noise_ratio

In addition to metrics classes, relevant handlers have been added for each metric inheriting from IterationMetric class.
A unittest file has also been added to cover the new features in testing.
Docstrings as well as relevant docs (metrics/handlers) have been updated with addition of the new metrics/handlers.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [x] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh -f -u --net --coverage`.
- [x] Quick tests passed locally by running `./runtests.sh --quick --unittests`.
- [x] In-line docstrings updated.
- [x] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

**Comment by masadcv** [comment]
Hi @Nic-Ma , @wyli this is now ready for review.

File: monai/metrics/regression.py [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+    Args:
+        reduction: {``"none"``, ``"mean"``, ``"sum"``, ``"mean_batch"``, ``"sum_batch"``,
+            ``"mean_channel"``, ``"sum_channel"``}
+            Define the mode to reduce computation result of 1 batch data. Defaults to ``"mean"``.
+
+    """
+
+    def __init__(self, reduction: Union[MetricReduction, str] = MetricReduction.MEAN) -> None:
+        super().__init__(reduction=reduction)
+        self.sq_func = partial(torch.pow, exponent=2.0)
```
Comment (wyli) [inline monai/metrics/regression.py:133]: not sure if this `partial` will work in a distributed evaluation case. could you follow the structure of https://github.com/Project-MONAI/MONAI/blob/45c6cf6117d2e25e88272a7001d84696e449a1cc/tests/test_handler_confusion_matrix_dist.py#L24
add a test file for RMSEMetric? the test requires two gpus on a single node.

Comment (masadcv) [reply monai/metrics/regression.py:133]: Sure thing, will add distributed tests for all metrics. The IterationMetric handlers for each new metric are in https://github.com/Project-MONAI/MONAI/blob/6ab5b9d40f62891bab36fb17e6c15d09fa189b17/monai/handlers/regression_metrics.py

For RMSEMetric the handler is here: https://github.com/Project-MONAI/MONAI/blob/6ab5b9d40f62891bab36fb17e6c15d09fa189b17/monai/handlers/regression_metrics.py#L87

Comment (masadcv) [reply monai/metrics/regression.py:133]: @wyli I have added distributed tests for all regression metrics handlers in the following file:

https://github.com/Project-MONAI/MONAI/blob/4e1db9c384026919753bca2253905cc158fe4924/tests/test_handler_regression_metrics_dist.py#L57

I tested these locally using 2 GPUs. They al pass without any changes to the metrics code. Can you please have another look and review this? From my side, it looks good to go, I would be happy to resolve any pending issues.

**Review by wyli — APPROVED** [review]
looks great, thank you!
